bgcval2/analysis_timeseries.py

```diff
@@ -139,6 +163,180 @@ def build_list_of_suite_keys(suites, debug=True):
+        return filepath
+    filepath = filepath.replace(lookingFor, new_value)
+    return filepath
+
+
+def parse_list_from_string(list1):
+    """
+    This tool takes a string and returns it as a list.
+    """
+    if isinstance(list1, list):
+       return list1
+
+    list1 = list1.replace('\t', ' ')
+    list1 = list1.replace(',', ' ')
+    list1 = list1.replace('  ', ' ')
+    list1 = list1.replace('\'', '')
+    list1 = list1.replace('\"', '')
+    list1 = list1.replace(';', '')
+    while list1.count('  ')>0:
+        list1 = list1.replace('  ', ' ')
+    #if len(list1)==0: return []
+    return list1.split(' ')
+
+
+def load_key_file(key, paths, jobID):
+    """
+    load_key_file:
+        takes an input file from key_files directory
+        and outputs a dictionairy.
+    """
+    output_dict = {}
+    print('load_key_file: checking {key}', key)
+    key_yml_path = os.path.join(os.path.join(paths.bgcval2_repo, 'key_files', ''.join([key.lower(),'.yml'])))
+    if os.path.exists(key_yml_path):
+        print('key_yml_path exists:', key_yml_path)
+    else:
+        print('key_yml_path does not exist:', key_yml_path)
+        raise FileNotFoundError
+
+    # Open yml file:
+    with open(key_yml_path, 'r') as openfile:
+        dictionary = yaml.safe_load(openfile)
+
+    if not dictionary or not isinstance(dictionary, dict):
+        print(f"Key Yaml file {key_yml_path} "
+              "is either empty or corrupt, please check its contents")
+        sys.exit(1)
+
+    # add jobID to dict:
+    dictionary['jobID'] = jobID
+
+    # Load basic fields: 
+    for field in ['name', 'units', 'dimensions', 'modelgrid']:
+        output_dict[field] = dictionary.get(field, None)
+        print('Adding', field,':', output_dict[field])
+
+    # Lists: 
+    for field in ['layers', 'regions', ]:
+        output_dict[field] = dictionary.get(field, '')
+        output_dict[field] = parse_list_from_string(output_dict[field])
+        print('Adding', field,':', output_dict[field])
+
+    # Metrics
+    if dictionary['dimensions'] in  [1,]:
+        metricList = ['metricless',]
+    else: 
+        metricList = ['mean', ] #'median', '10pc','20pc','30pc','40pc','50pc','60pc','70pc','80pc','90pc','min','max']
+    output_dict['metrics'] = dictionary.get('metrics', metricList)
+    output_dict['metrics'] = parse_list_from_string(output_dict['metrics'])
+
+    # Load Grid:
+    gridFile = dictionary.get('gridFile', paths.orcaGridfn)
+    output_dict['gridFile'] = list_input_files(gridFile, dictionary, paths)[0]
+
+    # load model or data specific parts: 
+    for m_or_d in ['model', 'data']:
+        md_vars = dictionary.get(''.join([m_or_d, '_vars']), False)
+        if m_or_d == 'data' and md_vars is False:
+            # Some analyses don't have observational data.  
+            continue
+       
+        if m_or_d == 'model' and md_vars is False:
+            print('What are you trying to analyse:', md_vars)
+            assert 0
```

Comment:
don't do assertions - raise errors or throw out warnings if it's less severe than an error